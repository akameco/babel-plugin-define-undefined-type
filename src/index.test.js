// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  plugin,
  snapshot: true,
  tests: [
    `type X = A`,
    `type X = Action`,
    `
    type Y = B;
    type X = A;
    `,
    `
    type X = A | B
    `,
    `
    type A = 'A'
    type X = A | B
    `,
    `
    type X = { type: ActionX };
    `,
    `
    type X = { type: A | B }
    `,
    `
    type X = { type: A }
    type Y = { type: B }
    type Action = X | Y
    `,
    `
    type Action = { type: ADD } | { type: INCREMENT }
    `
  ]
})
