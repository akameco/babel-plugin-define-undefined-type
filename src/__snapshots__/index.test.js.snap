// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. defined-undefined-type 1`] = `
"
type X = A

      ↓ ↓ ↓ ↓ ↓ ↓

type A = \\"A\\";
type X = A;
"
`;

exports[`2. defined-undefined-type 1`] = `
"
type X = Action

      ↓ ↓ ↓ ↓ ↓ ↓

type Action = \\"Action\\";
type X = Action;
"
`;

exports[`3. defined-undefined-type 1`] = `
"
type Y = B;
type X = A;

      ↓ ↓ ↓ ↓ ↓ ↓

type B = \\"B\\";
type Y = B;
type A = \\"A\\";
type X = A;
"
`;

exports[`4. defined-undefined-type 1`] = `
"
type X = A | B

      ↓ ↓ ↓ ↓ ↓ ↓

type A = \\"A\\";
type B = \\"B\\";
type X = A | B;
"
`;

exports[`5. defined-undefined-type 1`] = `
"
type A = 'A'
type X = A | B

      ↓ ↓ ↓ ↓ ↓ ↓

type A = 'A';
type B = 'B';
type X = A | B;
"
`;

exports[`6. defined-undefined-type 1`] = `
"
type X = { type: ActionX };

      ↓ ↓ ↓ ↓ ↓ ↓

type ActionX = \\"ActionX\\";
type X = { type: ActionX };
"
`;

exports[`7. defined-undefined-type 1`] = `
"
type X = { type: A | B }

      ↓ ↓ ↓ ↓ ↓ ↓

type A = \\"A\\";
type B = \\"B\\";
type X = { type: A | B };
"
`;

exports[`8. defined-undefined-type 1`] = `
"
type X = { type: A }
type Y = { type: B }
type Action = X | Y

      ↓ ↓ ↓ ↓ ↓ ↓

type A = \\"A\\";
type X = { type: A };
type B = \\"B\\";
type Y = { type: B };
type Action = X | Y;
"
`;

exports[`9. defined-undefined-type 1`] = `
"
type Action = { type: ADD } | { type: INCREMENT }

      ↓ ↓ ↓ ↓ ↓ ↓

type ADD = \\"ADD\\";
type INCREMENT = \\"INCREMENT\\";
type Action = { type: ADD } | { type: INCREMENT };
"
`;
